### ISSUE TITLE
Few observations during testing

### ISSUE THREAD

**Issue by ratheesh** [liuchengxu/vim-clap#22]
**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - NeoVim version: latest HEAD
 - vim-clap version: latest HEAD

**Describe the bug**
During testing of this plugin, I observed following.
- Current file position is reset to beginning of the file if clap window is opened and cancelled(by pressing ESC key)
- Adding to the above point, when an item(file based provider) on clip window is selected and executed, cursor is   reset to first line of the selected buffer. This is very annoying as user has to go back to desired buffer position manually.
- When clap floating window is displayed, current selected sign is not shown unless `C-j/k `are pressed.

**Comment by liuchengxu** [comment]
>Current file position is reset to beginning of the file if clap window is opened and cancelled(by pressing ESC key)

Can't reproduce with the minimal vimrc.

```vim
set nocompatible
set runtimepath^=~/.vim/plugged/vim-clap
syntax on
filetype plugin indent on
```

>Adding to the above point, when an item(file based provider) on clip window is selected and executed, cursor is reset to first line of the selected buffer. This is very annoying as user has to go back to desired buffer position manually.

I don't see the problem. vim-clap doesn't touch the cursor position when opening some file. Its behavior should be same with opening it with `:e [file]`. Let me know if it doesn't work like that.

>When clap floating window is displayed, current selected sign is not shown unless C-j/k are pressed.

This could be improved.

**Comment by liuchengxu** [comment]
@ratheesh Can you try #24? I have refined the current selection sign.

**Comment by ratheesh** [comment]
> Can't reproduce with the minimal vimrc.

I checked with the minimal vimrc. It is fine as you said. I will debug further on my side.

PR #24 works fine(neovim) for me. Thank you.

**Comment by ratheesh** [comment]
@liuchengxu I am able to reproduce file offset reset issue with below configuration(used Clap buffers command for testing).
```vim
set nocompatible
set runtimepath^=~/.vim/plugged/vim-clap
syntax on
filetype plugin indent on

function! ClapEnter() abort
    echom "ClapEnter"
endfunction " function

function! ClapExit() abort
    echom "ClapExit"
endfunction

augroup ClapGrp
    autocmd!
    autocmd User ClapOnEnter   execute ClapEnter()
    autocmd User ClapOnExit    execute ClapExit()
augroup END
```
Addition of user autocmd seems to be the culprit. Please let me know if something wrong with this.

**Comment by liuchengxu** [comment]
As I have said, vim-clap just uses `e` command to open the file, and does nothing else. Not so called _file offset reset issue_, that's expected. You could try any other plugins, like fzf.vim,  you'll always positioned to the first line when opening some file via `Files`.

I believe you are seeking for https://vi.stackexchange.com/questions/106/how-can-i-restore-the-cursor-position-when-opening-a-file. That should be in your vimrc, not an issue of a plugin.

**Comment by liuchengxu** [comment]
I have merged #24, let's see how it's going. Feel free to create another issue if you find new problem.

**Comment by ratheesh** [comment]
@liuchengxu Thank you. I will recheck on my side.
